### PR TITLE
Pract.07 : Add 01 | Todorov Nikolai

### PR DESCRIPTION
**Pull request by NIkoTod**
Todorov Nikolai SI 1 2MI8000043

### CODE REVIEW

**Comment by Justsvetoslavov** [comment]
@GeorgiTerziev02 @khatoni Can you take a look?

File: Seminars/Sem.07/Pract. 07/Task01/MyString.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    this->data = new char[size + 1];
+    strcpy(this->data, str);
+
+}
+MyString::MyString(const MyString &other){
+     copyFrom(other);
+}
+
+void MyString::free() {delete[] data;}
```
Comment (khatoni) [inline Seminars/Sem.07/Pract. 07/Task01/MyString.cpp:13]: Inconsistent with brackets { }

File: Seminars/Sem.07/Pract. 07/Task01/MyString.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+void MyString::copyFrom(const MyString &other) {
+
+    size = other.size;
+
+    data = new char[size + 1];
+    strcpy(data,other.data);
+
+}
+const char* MyString::c_str() {
```
Comment (khatoni) [inline Seminars/Sem.07/Pract. 07/Task01/MyString.cpp:22]: you do not change the object, so use const

File: Seminars/Sem.07/Pract. 07/Task01/MyString.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    void copyFrom(const MyString &other);
+
+
+public:
+
+    MyString(const char *str = "");
+    MyString(const MyString &str);
+    ~MyString();
+
+    const char* c_str();
```
Comment (khatoni) [inline Seminars/Sem.07/Pract. 07/Task01/MyString.h:19]: const

File: Seminars/Sem.07/Pract. 07/Task01/MyString.h [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    friend bool operator>(const MyString &lhs,const MyString &rhs);
+    friend bool operator<(const MyString &lhs,const MyString &rhs);
+    friend bool operator>=(const MyString &lhs,const MyString &rhs);
+    friend bool operator<=(const MyString &lhs,const MyString &rhs);
+
+    friend MyString operator*(const MyString &lhs,size_t num);
+    friend MyString operator*(size_t num,const MyString &rhs);
+
+    char operator[](size_t i);
+    char& operator[](size_t i) const;
```
Comment (khatoni) [inline Seminars/Sem.07/Pract. 07/Task01/MyString.h:46]: no const here

File: Seminars/Sem.07/Pract. 07/Task01/MyString.h [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    friend bool operator>(const MyString &lhs,const MyString &rhs);
+    friend bool operator<(const MyString &lhs,const MyString &rhs);
+    friend bool operator>=(const MyString &lhs,const MyString &rhs);
+    friend bool operator<=(const MyString &lhs,const MyString &rhs);
+
+    friend MyString operator*(const MyString &lhs,size_t num);
+    friend MyString operator*(size_t num,const MyString &rhs);
+
+    char operator[](size_t i);
```
Comment (khatoni) [inline Seminars/Sem.07/Pract. 07/Task01/MyString.h:45]: const here

File: Seminars/Sem.07/Pract. 07/Task01/MyString.cpp [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    return data;
+}
+size_t MyString::length() const {
+    return size;
+}
+
+bool MyString::empty() const {
+    if(size == 0)return true;
+    return false;
```
Comment (khatoni) [inline Seminars/Sem.07/Pract. 07/Task01/MyString.cpp:31]: return size==0

File: Seminars/Sem.07/Pract. 07/Task01/MyString.cpp [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    MyString res(lhs);
+    res+=rhs;
+    return res;
+
+}
+
+bool operator==(const MyString &lhs,const MyString &rhs){
+
+    if(strcmp(lhs.data,rhs.data) == 0)return true;
+    return false;
```
Comment (khatoni) [inline Seminars/Sem.07/Pract. 07/Task01/MyString.cpp:97]: one line

File: Seminars/Sem.07/Pract. 07/Task01/MyString.cpp [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+}
+bool operator!=(const MyString &lhs, const MyString &rhs) {
+    return !(lhs==rhs);
+}
+bool operator>(const MyString &lhs, const MyString &rhs) {
+    if(lhs.size > rhs.size)return true;
+    return false;
```
Comment (khatoni) [inline Seminars/Sem.07/Pract. 07/Task01/MyString.cpp:105]: one line

File: Seminars/Sem.07/Pract. 07/Task01/MyString.cpp [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    return data[i];
+}
+
+std::ostream& operator<<(std::ostream &os, const MyString &string) {
+    os << string.data;
+    return os;
+}
+std::istream &operator>>(std::istream &is, MyString &string) {
+
+    char* str = new char[8];
```
Comment (khatoni) [inline Seminars/Sem.07/Pract. 07/Task01/MyString.cpp:132]: why 8

**Review by khatoni — CHANGES_REQUESTED** [review]

**Pull request closed without merging**